src/chemistry/geoschem/cesmgc_history_mod.F90

```diff
@@ -0,0 +1,1191 @@
+#define _ASSERT(cond,msg) if(.not.cond) then; print *, "assertion error: ", Iam, __LINE__; call endrun("assertion error - look above - in cesmgc_history_mod.F90"); endif
+#define _Iam_(name) character(len=255) :: Iam=name
+#define __Iam__(name) integer :: STATUS; _Iam_(name)
+! Above are compatibility shorthands to avoid excessive divergence from
+! MAPL-based code. (hplin, 10/19/22)
+!------------------------------------------------------------------------------
+!                  GEOS-Chem Global Chemical Transport Model                  !
+!------------------------------------------------------------------------------
+!BOP
+!
+! !MODULE: cesmgc_history_mod.F90
+!
+! !DESCRIPTION: Module CESMGC\_History\_Mod interfaces between the CAM history
+!  component, the HISTORY.rc configuration file, and the GEOS-Chem State registry.
+!  This module is based off GCHP\_HistoryExports\_Mod originally developed by
+!  Lizzie Lundgren for GCHP.
+!\\
+!\\
+! !INTERFACE:
+!
+MODULE CESMGC_History_Mod
+!
+! !USES:
+!
+  USE DiagList_Mod
+  USE TaggedDiagList_Mod
+  USE ErrCode_Mod
+  USE Precision_Mod
+
+  USE cam_abortutils,      only : endrun
+
+  IMPLICIT NONE
+  PRIVATE
+!
+! !PUBLIC MEMBER FUNCTIONS:
+!
+  PUBLIC :: HistoryExports_SetServices
+  PUBLIC :: HistoryExports_SetDataPointers
+  PUBLIC :: CopyGCStates2Exports
+  PUBLIC :: Destroy_HistoryConfig
+!
+! !PRIVATE:
+!
+  PRIVATE :: Init_HistoryConfig
+  PRIVATE :: Init_HistoryExport
+  PRIVATE :: Init_HistoryExportsList
+  PRIVATE :: Append_HistoryExportsList
+  PRIVATE :: Check_HistoryExportsList
+  PRIVATE :: Print_HistoryExportsList
+ !
+! !PUBLIC TYPES
+!
+  ! History Configuration Object
+  TYPE, PUBLIC :: HistoryConfigObj
+
+     CHARACTER(LEN=255)                   :: ROOT ! TODO: needed?
```

Comment:
I added this member for GCHP but never actually used it. It's meant to be an ID for the HistoryConfigObj instance, but isn't used anywhere. In theory it could be used to help differentiate between multiple instances of the type.

Reply:
I think we ended up using one instance so I'll keep it as a stub here. Probably not needed.